src/languages/deno/deno.rs

```diff
@@ -129,17 +134,24 @@ impl LanguageImpl for Deno {
         filenames: &[&Path],
         store: &Store,
     ) -> Result<(i32, Vec<u8>)> {
-        let env_dir = hook.env_path().expect("Deno must have env path");
-        let bin_dir = env_dir.join("bin");
         let deno_dir = store.cache_path(CacheBucket::Deno);
-        let new_path = prepend_paths(&[&bin_dir]).context("Failed to join PATH")?;
+        let info = hook.install_info().expect("Deno hook must be installed");
+
+        // Use the toolchain path directly as the deno executable
+        let deno_bin = &info.toolchain;
 
-        let entry = hook.entry.resolve(None)?;
+        let entry = hook.entry.split()?;
         let run = async move |batch: &[&Path]| {
-            let mut output = Cmd::new(&entry[0], "deno run")
+            // Replace "deno" with the actual path to the installed deno binary
+            let command = if entry[0] == "deno" {
+                deno_bin.as_path()
+            } else {
+                Path::new(&entry[0])
+            };
```

Comment:
How about symlinking the deno executable into the hook env bin dir and prepending that bin dir to PATH, like we did in the `node` implementation? That might work better than this approach. For example, it would work for cases like `entry: ./run_deno.sh`.